src/librustdoc/html/render.rs

```diff
@@ -143,6 +144,55 @@ impl Impl {
     }
 }
 
+#[derive(Debug)]
+pub struct Error {
+    filename: Option<String>,
+    error: io::Error,
+}
+
+impl error::Error for Error {
+    fn description(&self) -> &str {
+        self.error.description()
+    }
+}
+
+impl Display for Error {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        match self.filename {
+            Some(ref file) => write!(f, "\"{}\": {}", file, self.error),
+            None => self.error.fmt(f),
+        }
+    }
+}
+
+impl Error {
+    pub fn from(e: io::Error) -> Error {
```

Comment:
I think this is basically the previous iteration but in a slightly more verbose fashion due to this. Basically all errors in rustdoc happen when working on some file, so if we want to attach the filename to the errors it should basically always be available to do so. Can you look into making filename not optional?